### PR TITLE
Enable GitHub Actions

### PR DESCRIPTION
**Pull request by maoe**
This PR

* fixes broken benchmark
* fixes or suppresses clippy warnings
* removes rust-toolchain
* and finally enables GitHub Actions

### CODE REVIEW

**Review by asayers — APPROVED** [review]
LGTM

File: rust-toolchain [hunk 1]
```diff
@@ -1 +0,0 @@
-1.44.1
```
Comment (Fuuzetsu) [inline rust-toolchain:1]: How come we're deleting this?

**Review by Fuuzetsu — APPROVED** [review]

File: rust-toolchain (hunk 1 above)
Comment (asayers) [reply rust-toolchain:1]: I guess it's unusual for a library to specify the toolchain.  It should work with a decent spread of recent rust versions.

Comment (maoe) [reply rust-toolchain:1]: As far as I understand, rust-toolchain is not supposed to be checked in in a library crate.

Comment (maoe) [reply rust-toolchain:1]: Unless you use a nightly toolchain

**Comment by maoe** [comment]
Thanks for the reviews. Merging.

**Pull request merged**